x-pack/plugins/observability/public/components/shared/alert_search_bar/alert_search_bar.tsx

```diff
@@ -126,13 +120,14 @@ export function ObservabilityAlertSearchBar({
           <EuiFlexItem grow={false}>
             <AlertsStatusFilter
               status={status}
-              onChange={(id) => {
-                setStatus(id as AlertStatus);
-              }}
+              onChange={(id) => onStatusChange(id as AlertStatus)}
```

Comment:
always avoid inline creation of functions and objects... they will trigger a re-render due to reference inequality.

Reply:
For example: https://amnsingh.medium.com/purecomponent-caveats-3a42b6e50e76

Reply:
I've read different articles about this topic and there are opposite views about it, such as this one:
https://reacttraining.com/blog/react-inline-functions-and-performance/
How do we check if re-rendering actually has a negative performance impact? (I assume the concern is performance)

Reply:
This is indeed a widely contested point.

AFAIK the performance hit is negligible (defining functions is considered cheap)

@maryam-saeidi if you want to know for sure, you could add `performance.now()`'s to the component and compare the different implementations. Testing in dev mode might give a skewed result, you want to test this in a production build.

Reply:
Even if the hit is "negligible", I haven't encountered an argument that defining inline is _preferable_, other than that's how it was written when the PR was created.

I personally pull them out into `const` for readability and to avoid waffling/considering it every time.

Reply:
@maryam-saeidi I just looked at the article you referenced... it's from 2017, and reflects class components, (e.g. `PureComponent`.

AFAIK, the strict equality check and re-render is *always* an issue with inline objects and props.  @CoenWarmer it's not about _defining_ functions, it's about the function causing a sub-component to re-render... which you can't really know how a re-render will affect children.

The rule of thumb still applies here.  I don't see an argument where defining inline _gains_ anything, where I see plenty of arguments why _not to_-- along with the cheapness of it, namely creating a `const`.

Reply:
@clintandrewhall Using inline functions is considered an acceptable practice, both in [the old](https://reactjs.org/docs/faq-functions.html#is-it-ok-to-use-arrow-functions-in-render-methods) and the [new React docs](https://beta.reactjs.org/learn/responding-to-events#adding-event-handlers). In this case, the component that had the inline function is not memoized, so unsure as to what value that would bring.

I'm however not against defining functions in a `const` as a rule of thumb.